src/backend/catalog/namespace.c

```diff
@@ -4554,31 +4548,16 @@ RemoveTempRelationsCallback(int code, Datum arg)
 		return;
 	}
 
+	/* If there is a temporary namespace, schedule its deletion */
 	if (OidIsValid(myTempNamespace))
+		GpScheduleSessionReset();
+
+	if (GpHasTempNamespaceForDeletion())
```

Comment:
may be add some comment before?

Reply:
may be "If we have a temporary namespace, delete it"

Reply:
For the `if (GpHasTempNamespaceForDeletion())` line? Okay